conans/client/build/mesonx.py

```diff
@@ -0,0 +1,375 @@
+            self._run_ninja_targets(targets=['install'],
+                                    args=self._filter_non_ninja_args(args=args))
+
+    def test(self, args=None):
+        if not self._conanfile.should_test:
+            return
+
+        args = args or []
+
+        self._run_meson_command(subcommand='test', args=args)
+
+    def is_configured(self):
+        _, build_dir = self._get_resolved_dirs()
+        if self._version >= '0.50.0':
+            return (Path(build_dir) / 'meson-info' / 'meson-info.json').exists()
+        else:
+            return (Path(build_dir) / 'meson-private' / 'coredata.dat').exists()
+
+    def _ss(self, setname):
+        """safe setting"""
+        return self._conanfile.settings.get_safe(setname)
+
+    def _so(self, setname):
+        """safe option"""
+        return self._conanfile.options.get_safe(setname)
+
+    def _validate_ninja_usage_and_warn_agnostic_method_unavailable(self, minimum_version: str):
+        if self.backend != 'ninja':
+            raise ConanException('This method is not implemented yet for `{}` backend. Change your backend to `ninja` or update your `meson`.\n'.format(self.backend) +
+                                 'Minimum required `meson` version is {}'.format(minimum_version))
+        self._conanfile.output.warn(
+            'Backend agnostic version of this method is not supported on this `meson` version. Using `ninja` directly instead.\n'
+            'Minimum required `meson` version is {}'.format(minimum_version))
+
+    def _filter_non_ninja_args(self, args=None, ninja_args=None):
+        if not args:
+            return []
+
+        args = args or []
+        ninja_args = ninja_args or []
+
+        filtered_args = []
+        remaining_args = []
+        for a in args:
+            for n in ninja_args:
+                if (n.endswith('*') and a.startswith(n[:-1])) or a == n:
+                    filtered_args += [a]
+                else:
+                    remaining_args += [a]
+
+        if remaining_args:
+            self._conanfile.output.warn(
+                'The following `meson` args are not supported by `ninja` and will be ignored: "{}"'.format(' '.join(remaining_args)))
+
+        return filtered_args
+
+    def _get_resolved_dirs(self, source_subdir=None):
+        """
+        Returns (resolved_source_dir, resolved_build_dir)
+        """
+        source_dir = self._conanfile.source_folder
+        if source_subdir:
+            source_dir = str(Path(self._conanfile.source_folder) / source_subdir)
+
+        build_dir = self._conanfile.build_folder
+        if self._build_subdir:
+            build_dir = str(Path(self._conanfile.build_folder) / _build_subdir)
+        return (source_dir, build_dir)
+
+    def _get_default_options(self) -> dict:
+        options = dict()
+        if self._conanfile.package_folder:
+            options['prefix'] = self._conanfile.package_folder
+        options['libdir'] = DEFAULT_LIB
```

Comment:
Uhm, but won't this fail during local dev flow? For example, a user invokes `conan configure . -bf=local_folder -sf=some_folder`, but, since `package_folder` is not available, this command will raise.

Reply:
It is better to provide partial functionality at the moment and raise if something is not implemented yet (for example, the local flow), than provide flaky or hardcoded assumptions, like using DEFAULT_BIN and DEFAULT_LIB directories, that some users will start relying on that behavior, and later will be broken when we change it.

Reply:
The problem is that nobody will use (and test) this feature if it's doesn't work in the very basic and common scenario (esp. when the current helper does provide such functionality). Maybe add a note that "local development flow is highly experimental and might be/will be changed in the future updates"?
^ this about about `prefix` only. I don't have much objections to removing `DEFAULT_BIN` and friends (I don't see why it's bad though, esp. since conan itself expects a fixed layout when searching for dep binaries).

Reply:
Note: removed DEFAULT_BIN and friends, but kept `prefix` for now.